src/shared/components/query/QueryStore.ts

```diff
@@ -728,47 +744,52 @@ export class QueryStore
+			genetic_profile_ids_PROFILE_MRNA_EXPRESSION: this.getSelectedProfileIdFromGeneticAlterationType("MRNA_EXPRESSION", queryParams.selectedProfileIds),
+			genetic_profile_ids_PROFILE_METHYLATION: this.getSelectedProfileIdFromGeneticAlterationType("METHYLATION", queryParams.selectedProfileIds) ||
+														this.getSelectedProfileIdFromGeneticAlterationType("METHYLATION_BINARY", queryParams.selectedProfileIds),
+			genetic_profile_ids_PROFILE_PROTEIN_EXPRESSION: this.getSelectedProfileIdFromGeneticAlterationType("PROTEIN_LEVEL", queryParams.selectedProfileIds),
+			Z_SCORE_THRESHOLD: queryParams.zScoreThreshold,
+			RPPA_SCORE_THRESHOLD: queryParams.rppaScoreThreshold,
+			data_priority: this.getDataTypePriorityCode(queryParams.dataTypePriority),
+			case_set_id: queryParams.selectedSampleListId || '-1', // empty string won't work
+			case_ids: queryParams.caseIds,
+			gene_list: queryParams.geneQuery || ' ', // empty string won't work
+			tab_index: this.forDownloadTab ? 'tab_download' : 'tab_visualize' as any,
+			transpose_matrix: this.transposeDataMatrix ? 'on' : undefined,
+			Action: 'Submit',
+		};
 
-			// Remove params with no value, because they may cause problems.
-			// For example, the server will always transpose if transpose_matrix is present, no matter the value.
-			for (let key in params)
-				if (!(params as any)[key])
-					delete (params as any)[key];
+		// Remove params with no value, because they may cause problems.
+		// For example, the server will always transpose if transpose_matrix is present, no matter the value.
+		for (let key in params)
+			if (!(params as any)[key])
+				delete (params as any)[key];
 
-			if (this.selectedStudyIds.length != 1)
-			{
-				let studyIds = this.selectedStudyIds;
-				if (!studyIds.length)
-					studyIds = this.cancerStudies.result.map(study => study.studyId);
+		if (queryParams.selectedStudyIds.length != 1)
+		{
+			let studyIds = queryParams.selectedStudyIds;
+			if (!studyIds.length)
+				studyIds = this.cancerStudies.result.map(study => study.studyId);
 
-				const hash = `crosscancer/overview/${params.data_priority}/${encodeURIComponent(params.gene_list)}/${encodeURIComponent(studyIds.join(','))}`;
-				return {
-					pathname: `cross_cancer.do#${hash}`,
-					query: Object.assign({ cancer_study_list: studyIds.join(",")}, params),
-				};
-			}
+			const hash = `crosscancer/overview/${params.data_priority}/${encodeURIComponent(params.gene_list)}/${encodeURIComponent(studyIds.join(','))}`;
+			return {
+				pathname: `cross_cancer.do#${hash}`,
+				query: Object.assign({ cancer_study_list: studyIds.join(",")}, params),
+			};
+		}
+
+		return {pathname: 'index.do', query: params};
+	}
 
-			return {pathname: 'index.do', query: params};
+	readonly asyncUrlParams = remoteData({
+		await: () => [this.asyncCustomCaseSet, this.geneticProfiles],
+		invoke: async () => {
+			return this.getUrlParams(this);

```

Comment:
nope, initialParams is only in addGenesToQuery, if you look at the type of initialParams, its a Pick type of QueryStore :)